src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -253,6 +253,9 @@ public FeatureDataSource(final FeatureInput<T> featureInput, final int queryLook
         } else {
             throw new GATKException("Found a feature input that was neither GenomicsDB or a Tribble AbstractFeatureReader.  Input was " + featureInput.toString() + ".");
         }
+        if (!hasIndex && TribbleIndexedFeatureReader.hasBlockCompressedExtension(featureInput.getFeaturePath())) {
+            throw new GATKException("Found a feature input that was block compressed but lacked an index file.  Input was " + featureInput.toString() + ".");
```

Comment:
I agree with @magicDGS on both points. This should be a `UserException.MissingIndex` (which you should create), and include a note that "The requirement that all block-compressed inputs have an index may be relaxed in the future." The error message should also mention that the tool `IndexFeatureFile` can be used to create the necessary index.

Reply:
You should also add a comment above this if statement with a link to the relevant bug on github, and a brief explanation of why this constraint exists.

Reply:
done